### DISCUSSION TITLE
How can I toggle show/hide whitespace and indentation lines?

### DISCUSSION THREAD

**Discussion by andradei** [LazyVim/LazyVim#341] [Q&A] [answered]
I'd like to to show/hide spaces, tabs, etc via a toggle command.
Same for indentation guides (lines) that come with LazyVim.

How do I do that?

**Comment by yusufaktepe** [comment, marked as answer]
Configure these plugins to be disabled when the `list` option is unset:
```lua
  {
    "lukas-reineke/indent-blankline.nvim",
    opts = {
      disable_with_nolist = true,
    },
  },

  {
    "echasnovski/mini.indentscope",
    cond = vim.opt.list:get(),
  },
```

Toggle function:
```lua
local function toggle_list()
  if vim.opt.list:get() then
    vim.cmd("setlocal nolist")
    vim.b.miniindentscope_disable = true
  else
    vim.cmd("setlocal list")
    vim.b.miniindentscope_disable = false
  end
end

vim.keymap.set("n", "<leader>uL", toggle_list, { desc = "Toggle listchars" })
```

**Comment by notDavid** [comment]
Hi there @yusufaktepe , thank you for your provided solution;

Noob question here, I can't get it to work (anymore?), if you could advise how i can fix it i would be very grateful :-)

I put the function in `config/options.lua` and the plugin settings in `plugins/mysettings.lua`. Here's the error i get:

```log
Error  08:25:21 notify.error lazy.nvim Failed to run `config` for indent-blankline.nvim

.../share/nvim/lazy/indent-blankline.nvim/lua/ibl/utils.lua:18: 'disable_with_nolist' is not a valid key of ibl.config

# stacktrace:
  - /indent-blankline.nvim/lua/ibl/utils.lua:18 _in_ **validate**
  - /indent-blankline.nvim/lua/ibl/config.lua:112 _in_ **validate_config**
  - /indent-blankline.nvim/lua/ibl/config.lua:273 _in_ **set_config**
  - /indent-blankline.nvim/lua/ibl/init.lua:58 _in_ **setup**
  - /LazyVim/lua/lazyvim/util/plugin.lua:93 _in_ ****
  - vim/_editor.lua:0

```

Thank you!

**Reply by dpetka2001** [reply]
Put the following in your `/lua/config/keymaps.lua`
```lua
local function toggle_list()
  if vim.opt.list:get() then
    vim.cmd("setlocal nolist")
    vim.cmd("IBLDisable")
    vim.b.miniindentscope_disable = true
  else
    vim.cmd("setlocal list")
    vim.cmd("IBLEnable")
    vim.b.miniindentscope_disable = false
  end
end

vim.keymap.set("n", "<leader>uL", toggle_list, { desc = "Toggle listchars" })
```
Remove the settings you put in the plugins. `indent-blankline` had breaking changes in version 3.x and I don't see this key in the configuration any more.

**Reply by notDavid** [reply]
Great, thank you so much!  🤗️

**Reply by notDavid** [reply]
@dpetka2001 ...strange, i'm suddenly getting this error. Any idea? 🙏🏼

```
E5108: Error executing lua: vim/_editor.lua:0: nvim_exec2(), line 1: Vim:E492: Not an editor command: IBLEnable
stack traceback:
	[C]: in function 'nvim_exec2'
	vim/_editor.lua: in function 'cmd'
	/Users/david/.config/nvim/lua/config/options.lua:31: in function </Users/david/.config/nvim/lua/config/options.lua:28>
```

**Reply by dpetka2001** [reply]
The plugin `indent-blankline.nvim` was moved into Extras and is not enabled by default. `snacks.nvim` has an `indent` functionality and is its replacement.

You have to enable the Extra `ui.indent-blankline` from `:LazyExtras` to have the previous functionality. Or you should read the docs of `snacks.nvim` to see if you can have the same functionality with Snacks.

This change was made around 7 months ago. You should read the [changes news](https://www.lazyvim.org/news) if you haven't used LazyVim for a long time.

PS: also there's a keymap `<leader>ug` to toggle on/off indentation guides for convenience.

**Reply by notDavid** [reply]
@dpetka2001 Thank you for the clarifying / the details, much appreciated.
Yea, it's almost mission impossible to keep up with all the changes... especially for "vim noobs" ;-)

**Reply by notDavid** [reply]
For future reference...: `vim.b.snacks_indent = false`

```lua
local function toggle_list()
  if vim.opt.list:get() then
    vim.cmd("setlocal nolist")
    vim.b.snacks_indent = true
    vim.b.miniindentscope_disable = true
  else
    vim.cmd("setlocal list")
    vim.b.snacks_indent = false
    vim.b.miniindentscope_disable = false
  end
end

vim.keymap.set("n", "<leader>uL", toggle_list, { desc = "Toggle listchars" })
```
